### PR TITLE
Fix cubeviz unit conversion

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

When creating the new Spectrum1D with different units, the old WCS was preserved and inconsistent with the spectral axis. I'd almost call that a Spectrum1D bug *except* that this is ``_copy`` and private API. I'd highly recommend that this be done via public API in future but for now dropping the original WCS does the trick since the WCS is then produced from the provided spectral axis.

Screenshot from 2021-09-16 21-11-25

Fixes https://github.com/spacetelescope/jdaviz/issues/886

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py [hunk 1]
```diff
@@ -294,6 +294,7 @@ def process_unit_conversion(self, spectrum, new_flux=None, new_spectral_axis=Non
 
         # Create new spectrum with new units.
         converted_spectrum = spectrum._copy(flux=set_flux_unit,
+                                            wcs=None,
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:297]: Pure magic! ⭐⭐⭐⭐⭐

**Review by pllim — APPROVED** [review]

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py (hunk 1 above)
Comment (javerbukh) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:297]: 😂

**Review by javerbukh — APPROVED** [review]
Thank you!

**Review by rosteen — APPROVED** [review]
Agreed that this should really just be initializing a new `Spectrum1D` from the properties here rather than using `_copy`, but this is a good enough fix to get in for the build deadline.

**Pull request merged**